Platon/backend/app/search_engine/forms.py

```diff
@@ -34,4 +34,15 @@ class WorkspaceSearchForm(Form):
 ws_search_parser.add_argument('event_filter',type=str,help="Give Acronym",location='args')
 ws_search_parser.add_argument('page',type=int,help="Page ID(0-Indexed)",location='args')
 ws_search_parser.add_argument('per_page',type=int,help="Number of Records in a Page",location='args')
-ws_search_parser.add_argument('auth_token', type=str,help="Authantication Token(If registered)",location='headers')
\ No newline at end of file
+ws_search_parser.add_argument('auth_token', type=str,help="Authantication Token(If registered)",location='headers')
+
+class UpcomingEventsSearchForm(Form):
+    search_query = StringField("search_query",validators=[validators.DataRequired()])
+    page = IntegerField("page")
+    per_page = IntegerField("per_page")
+
```

Comment:
1.1.6.2 Users and guests shall be able to do an advanced search on only public workspaces, upcoming events, and any profile by specifying filters such as founder of the project, starting date, submission deadline, research area/topic and labels.

According to this requirement, there must be a filtering mechanism on Upcoming Event Search which filters the events using their deadlines and starting dates. Can you add such a feature?

Reply:
Added them on my last commit.